src/components/accordion/__tests__/CdrAccordion.spec.js

```diff
@@ -1,212 +1,158 @@
+    default: 'This is some slot text.',
+    label: 'label',
+  },
+}
 
 describe('CdrAccordion', () => {
-  it('renders correctly', () => {
-    const wrapper = mount(CdrAccordion, {
-      propsData: {
-        id: 'test',
-        level: '2',
-      },
-      slots: {
-        default: 'This is some slot text.',
-        label: 'label'
-      },
+  describe('when mounted', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = mount(CdrAccordion, { ...baseComponentPattern });
+    });
+
+    it('renders correctly', () => {
+      expect(wrapper.element).toMatchSnapshot()
     });
-    expect(wrapper.element).toMatchSnapshot()
-  });
+  })
 
-  it('renders correctly unwrapped', async () => {
-    const wrapper = mount(CdrAccordion, {
-      propsData: {
-        id: 'test',
-        level: '2',
-      },
-      slots: {
-        default: 'This is some slot text.',
-        label: 'label'
-      },
-      global: {
-        provide: {
-          unwrap: {
-            isUnwrapped: {
-              value: true
+  describe('when unwrapped', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      const unwrappedComponent = {
+        ...baseComponentPattern,
+        global: {
+          provide: {
+            unwrap: {
+              isUnwrapped: {
+                value: true
+              }
             }
-          }
+          },
         }
       }
+      wrapper = mount(CdrAccordion, unwrappedComponent);
     });
-    await wrapper.vm.$nextTick();
-    expect(wrapper.element).toMatchSnapshot()
-  });
-
-  // TODO add more checks here
-  it('meets a11y requirements', async () => {
-    const wrapper = shallowMount(CdrAccordion, {
-      propsData: {
-        id: 'test',
-        level: '2',
-      },
-      slots: {
-        default: 'This is some slot text.',
-        label: 'label',
-      },
+    it('renders correctly', async () => {
+      expect(wrapper.element).toMatchSnapshot()
     });
-    const button = wrapper.find('#test');
-    const contentArea = wrapper.find('.cdr-accordion__content');
-    expect(button.classes()).toContain('js-cdr-accordion-button');
-    // closed state
-    expect(button.attributes('aria-expanded')).toBe('false');
-    expect(button.attributes('aria-controls')).toBe(`${wrapper.vm.id}-collapsible`);
-
-    // opened state
-    button.trigger('click');
-    wrapper.setProps({ opened: true }); // fake the opening logic
-    await wrapper.vm.$nextTick();
-    expect(button.attributes('aria-expanded')).toBe('true');
-    expect(contentArea.attributes('aria-hidden')).toBe('false');
   })
 
-  describe('accordion data setup', () => {
-    it('sets maxHeight when starting closed', () => {
-      const wrapper = shallowMount(CdrAccordion, {
-        propsData: {
-          id: 'test',
-          level: '2',
-          opened: false,
-        },
-        slots: {
-          default: 'This is some slot text.',
-          label: 'label',
-        },
+  describe('when shallow mounted', () => {
```

Comment:
'when shallow mounted' feels odd to me as a describe since that's a specific pattern for a test and not how this would be used in real life

Reply:
That makes sense. After looking into the difference between `mount` and `shallowMount` I think it's fair to describe these a component unit tests.